content/influxdb/cloud-dedicated/query-data/execute-queries/troubleshoot.md

```diff
@@ -0,0 +1,274 @@
+description: >
+  Understand responses and troubleshoot errors encountered when querying {{% cloud-name %}} with Flight+gRPC and Arrow Flight clients.
+weight: 401
+menu:
+  influxdb_cloud_dedicated:
+    name: Understand Flight responses
+    parent: Execute queries
+influxdb/cloud-dedicated/tags: [query, sql, influxql]
+---
+
+Learn how to handle responses and troubleshoot errors encountered when querying {{% cloud-name %}} with Flight+gRPC and Arrow Flight clients.
+
+<!-- TOC -->
+
+- [InfluxDB Flight responses](#influxdb-flight-responses)
+  - [Stream](#stream)
+  - [Schema](#schema)
+    - [Example](#example)
+  - [RecordBatch](#recordbatch)
+  - [InfluxDB status and error codes](#influxdb-status-and-error-codes)
+  - [Troubleshoot errors](#troubleshoot-errors)
+    - [Internal Error: Received RST_STREAM](#internal-error-received-rst_stream)
+    - [Internal Error: stream terminated by RST_STREAM with NO_ERROR](#internal-error-stream-terminated-by-rst_stream-with-no_error)
+    - [Invalid Argument: Invalid ticket](#invalid-argument-invalid-ticket)
+    - [Unauthenticated: Unauthenticated](#unauthenticated-unauthenticated)
+    - [Unauthorized: Permission denied](#unauthorized-permission-denied)
+    - [FlightUnavailableError: Could not get default pem root certs](#flightunavailableerror-could-not-get-default-pem-root-certs)
+
+## InfluxDB Flight responses
+
+{{% cloud-name %}} provides an InfluxDB-specific Arrow Flight RPC and Flight SQL service that uses gRPC Remote Procedure Calls (gRPC) to transport data in Arrow format.
+Flight defines a set of [RPC methods](https://arrow.apache.org/docs/format/Flight.html#rpc-methods-and-request-patterns) that servers and clients can use to exchange information.
+Flight SQL uses Flight RPC and defines additional methods to query database metadata, execute queries, and manipulate prepared statements.
+To learn more about Flight SQL, see [Introducing Apache Arrow Flight SQL: Accelerating Database Access](https://arrow.apache.org/blog/2022/02/16/introducing-arrow-flight-sql/).
+
+To query data or retrieve information about data stored in {{% cloud-name %}}, a Flight client (for example, `influx3` or an InfluxDB v3 client library) sends a request that calls an InfluxDB Flight RPC or Flight SQL service method.
+For example, if you call the `influxdb3-python` Python client library `InfluxDBClient3.query()` method, the client in turn calls the `pyarrow.flight.FlightClient.do_get()` method that passes a Flight ticket containing your credentials and query to InfluxDB's Flight [`DoGet(FlightCallOptions, Ticket)` method](https://arrow.apache.org/docs/cpp/api/flight.html#_CPPv4N5arrow6flight12FlightClient5DoGetERK17FlightCallOptionsRK6Ticket).
+
+InfluxDB responds with one of the following:
+
+- A [stream](#stream) in Arrow IPC streaming format
+- An [error status code](#influxdb-error-codes) and an optional `details` field that contains the status and a message that describes the error
```

Comment:
Does it only ever respond with error status codes, or does it also respond with success status codes?